### PR TITLE
Story/quad 196

### PR DESCRIPTION
**Pull request by suprajkolluri**

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java [hunk 1]
```diff
@@ -346,6 +346,27 @@ public INetwork getNetwork(String networkId) throws QuadrigaStorageException {
             networkList = networkmapper.getListOfNetworksForUser(user);
         } catch (QuadrigaStorageException e) {
             logger.error("Something went wrong in DB", e);
+            throw e;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java:349]: why do you catch the exception just to throw it?

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java [hunk 2]
```diff
@@ -346,6 +346,27 @@ public INetwork getNetwork(String networkId) throws QuadrigaStorageException {
...
+    @Override
+    @Transactional
+    public List<INetwork> getApprovedNetworkList() throws QuadrigaStorageException {
+        List<INetwork> networkList = new ArrayList<INetwork>();
+
+        try {
+            networkList = networkmapper.getListOfApprovedNetworks();
+        } catch (QuadrigaStorageException e) {
+            logger.error("Something went wrong in DB", e);
+            throw e;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java:369]: same here

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 3]
```diff
@@ -54,20 +54,34 @@ public ITransformedNetwork getTransformedNetwork(String networkId, String versio
...
 
         List<INetworkNodeInfo> networkTopNodesList = null;
 
         try {
             networkTopNodesList = networkManager.getNetworkTopNodes(networkId);
         } catch (QuadrigaStorageException e) {
             logger.error("DB Error while getting network top nodes", e);
-            return null;
+            throw e;
         }
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java:65]: and here

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java [hunk 4]
```diff
@@ -54,20 +54,34 @@ public ITransformedNetwork getTransformedNetwork(String networkId, String versio
...
 
-        return networkJSon;
+        List<INetworkNodeInfo> networkTopNodesList = new ArrayList<INetworkNodeInfo>();
+        for(INetwork curnetwork: networkList){
+            try {
+                List<INetworkNodeInfo> curTopNodesList = networkManager.getNetworkTopNodes(curnetwork.getNetworkId());
+                networkTopNodesList.addAll(curTopNodesList) ;
+            } catch (QuadrigaStorageException e) {
+                logger.error("DB Error while getting network top nodes", e);
+                throw e;
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java:80]: and here

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java [hunk 5]
```diff
@@ -26,26 +26,27 @@
...
+            throws Exception {
         Set<TextOccurance> occurances = networkManager.getTextsForConceptId(conceptId, ETextAccessibility.PUBLIC);
         JSONArray projectTexts = new JSONArray();
         JSONArray otherProjectsTexts = new JSONArray();
-        
+
         for (TextOccurance occur : occurances) {
-            if (occur.getProject().getUnixName().equals(projectUnix.trim())) {
+            if (occur.getProject().getUnixName().equals(projectUnix.trim())
+                    || occur.getProject().getUnixName().equals("unknown")) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/open/NodeInfoController.java:45]: that's not very good design. not very maintainable or intuitive. just make the projectUnix name optional and make sure the projects that are retrieved are public.

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java (hunk 1 above)
Comment (suprajkolluri) [reply Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkManager.java:349]: I just wanted to log the exception

**Pull request closed without merging**